registry/protocol/protocol.go

```diff
@@ -338,10 +338,10 @@ func setProviderUrl(regURL *common.URL, providerURL *common.URL) {
 }
 
 func GetProtocol() protocol.Protocol {
```

Comment:
This method I had reverted to its origin version. You can merge master branch and there is not necessary to revert it again.

Reply:
Yes, I know this, but I don't know how to merge your code into the development branch. This problem affects my test, so I added your recovered code to my branch